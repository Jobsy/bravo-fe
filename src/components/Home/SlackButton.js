import React from 'react';

const SlackButton = () => {
  const clientId = process.env.REACT_APP_CLIENT_ID;
  const redirectURI = process.env.REACT_APP_REDIRECT_URI;
  const scope = 'identity.basic,identity.email,identity.team,identity.avatar';
  const url = `https://slack.com/oauth/authorize?scope=${scope}&client_id=${clientId}&redirect_uri=${redirectURI}`;

  return (
    <a href={url}>
      <img
        alt="Sign in with Slack"
        height="40"
        width="172"
        src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
        srcSet="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
      />
    </a>
  );
};

export default SlackButton;
