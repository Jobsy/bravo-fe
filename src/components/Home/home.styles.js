import styled from 'styled-components';
import { PageTitle, BodyText } from '../../styling/atoms/Fonts';

export const Container = styled.div`
  margin: auto;
  font-family: 'Lato', sans-serif;
  font-size: 62.5%;
  max-width: 1440px;
  max-height: 1000px;
`;

export const Main = styled.div`
  width: 900px;
  height: 450px;
  margin: 60px auto;
  justify-content: space-between;
`;

export const Intro = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 200px;
  margin-bottom: 100px;
`;

export const Caption = styled.span`
  font-weight: bold;
  margin-bottom: 10px;
  font-size: 48px;
  line-height: 22px;
`;

export const IntroText = styled.span`
  text-align: center;
  width: 530px;
  font-size: 20px;
  line-height: 24px;
  font-weight: 300;
  color: #414a53;
`;

export const StyledPageTitle = styled(PageTitle)`
  margin: 0 auto;
`;

export const StyledExtraInfo = styled(BodyText)`
  text-align: center;
  margin: 20px auto 30px auto;
  width: 530px;
  height: 45px;
`;
