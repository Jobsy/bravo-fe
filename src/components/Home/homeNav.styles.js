import styled from 'styled-components';
import colors from '../../styling/variables/UIColors';

export const HomeNavContainer = styled.nav`
  position: fixed;
  width: 100%;
  top: 0;
  left: 0;
  margin: auto;
  display: flex;
  justify-content: space-between;
  padding: 25px 0;
  background-color: ${colors.light};

  @media (max-width: 850px) {
    padding: 15px;
  }
`;

export const NavContainer = styled.nav`
  width: 980px;
  margin: auto;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const LogoContainer = styled.div`
  display: flex;
`;

export const Logo = styled.img`
  width: 48px;
  height: 48px;
`;

export const NavText = styled.span`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  font-weight: 600;
  font-size: 32px;
  line-height: 38px;
  margin-left: 8px;

  @media (max-width: 380px) {
    display: none;
  }
`;
