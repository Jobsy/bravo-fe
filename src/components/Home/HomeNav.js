import React from 'react';

import { NavContainer, LogoContainer, Logo, NavText, HomeNavContainer } from './homeNav.styles';
import logoImg from '../../assets/bravo-logo.png';
import { SignInWithSlack } from '../SlackButton';

const HomeNav = () => {
  return (
    <HomeNavContainer>
      <NavContainer>
        <LogoContainer>
          <Logo src={logoImg} alt="Bravo logo" />
          <NavText>bravo</NavText>
        </LogoContainer>
        <SignInWithSlack />
      </NavContainer>
    </HomeNavContainer>
  );
};

export default HomeNav;
