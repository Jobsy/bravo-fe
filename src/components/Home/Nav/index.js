import React from 'react';

import { SignInWithSlack } from '../SlackButton';
import { NavContainer, LogoContainer, Logo, NavText } from './nav.styles';
import logoImg from '../../../assets/bravo-logo.png';

const Nav = () => {
  return (
    <NavContainer>
      <LogoContainer>
        <Logo src={logoImg} />
        <NavText>bravo</NavText>
      </LogoContainer>
      <SignInWithSlack />
    </NavContainer>
  );
};

export default Nav;
