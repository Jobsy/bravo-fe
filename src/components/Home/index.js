import React from 'react';

import { Container, Main, Intro, StyledPageTitle, StyledExtraInfo } from './home.styles';
import { AddToSlack } from '../SlackButton';
import HomeCards from '../Cards/HomeCards';
import HomeNav from '../HomeNav';

const Home = () => {
  return (
    <Container>
      <HomeNav />
      <Main>
        <Intro>
          <StyledPageTitle fontWeight="bold">when the job is done, say bravo</StyledPageTitle>
          <StyledExtraInfo>
            bravo enables team members to easily acknowledge each other with one slack command
          </StyledExtraInfo>
          <AddToSlack />
        </Intro>
        <HomeCards />
      </Main>
    </Container>
  );
};
export default Home;
