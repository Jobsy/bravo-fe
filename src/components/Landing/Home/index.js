import React, { useState, useEffect } from 'react';
import axios from 'axios';
import qs from 'querystring';

import { Container, Main, Intro, Caption, IntroText } from './home.styles';

import Cards from '../Cards';
import Nav from '../Nav';
import { AddToSlack } from '../SlackButton';
import Success from '../Success/index';

const Home = props => {
  const {
    location: { search },
  } = props;
  // Parse slack temporary code in URL
  const parsed = qs.parse(search);
  const { '?code': code } = parsed;
  const [addBotSuccess, setAddBotSuccess] = useState(false);

  useEffect(() => {
    const reqBody = {
      client_id: process.env.REACT_APP_CLIENT_ID,
      client_secret: process.env.REACT_APP_CLIENT_SECRET,
      code,
    };

    // Get slack access token
    if (code) {
      axios
        .post('https://slack.com/api/oauth.access', qs.stringify(reqBody), {
          headers: {
            'Content-Type': 'application/x-www-form-urlencoded',
          },
        })
        .then(res => {
          /* 
            @ res.data - contains access token from slack 
                  - contains 'bot' if Add to Slack was Clicked
                  - contains 'user' if Log in with Slack was clicked
          */
          // set to true to display success message
          console.log(res);
          setAddBotSuccess(true);
        });
    }
  }, [code]);

  return (
    <Container>
      <Nav />
      <Main>
        <Intro>
          <Caption>when the job is done, say bravo</Caption>
          <IntroText>
            bravo enables team members to easily acknowledge each other with one slack command
          </IntroText>
          {addBotSuccess ? <Success /> : <AddToSlack />}
        </Intro>
        <Cards />
      </Main>
    </Container>
  );
};
export default Home;
