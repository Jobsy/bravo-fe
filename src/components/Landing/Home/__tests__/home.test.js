import React from 'react';
import * as rtl from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';

import Home from '../index';

afterEach(rtl.cleanup);
describe('Home Component', () => {
  it('Should render home component', () => {
    const { container } = rtl.render(<Home location={{ search: '' }} />);
    expect(container).toMatchSnapshot();
  });
});
