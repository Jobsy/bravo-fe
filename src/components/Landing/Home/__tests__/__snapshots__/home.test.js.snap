// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should render home component 1`] = `
<div
  class="sc-bdVaJa kxiCan"
>
  <nav
    class="sc-VigVT kzsZlb"
  >
    <div
      class="sc-jTzLTM eoQMeB"
    >
      <img
        class="sc-fjdhpX nsimD"
        src="bravo-logo.png"
      />
      <span
        class="sc-jzJRlG dHkEnb"
      >
        Bravo
      </span>
    </div>
    <a
      href="https://slack.com/oauth/authorize?scope=identity.basic,identity.email,identity.team,identity.avatar&client_id=711954789234.734180204582"
    >
      <img
        alt="Sign in with Slack"
        height="40"
        src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
        srcset="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
        width="172"
      />
    </a>
  </nav>
  <div
    class="sc-bwzfXH jZeGHR"
  >
    <div
      class="sc-htpNat jWhSrG"
    >
      <span
        class="sc-bxivhb MyuFa"
      >
        when the job is done, say bravo
      </span>
      <span
        class="sc-ifAKCX bsCEmw"
      >
        bravo enables team members to easily acknowledge each other with one slack command
      </span>
      <a
        href="https://slack.com/oauth/authorize?client_id=711954789234.734180204582&scope=commands,chat:write:bot,chat:write:user,im:write,im:read,im:history,users:read,channels:read"
      >
        <img
          alt="Add to Slack"
          height="40"
          src="https://platform.slack-edge.com/img/add_to_slack.png"
          srcset="https://platform.slack-edge.com/img/add_to_slack.png 1x, https://platform.slack-edge.com/img/add_to_slack@2x.png 2x"
          width="139"
        />
      </a>
    </div>
    <div
      class="sc-bZQynM ktJREs"
    >
      <div
        class="sc-gzVnrw gQetWj"
      >
        <img
          class="sc-iwsKbI hHToOU"
          src="clap-icon.png"
        />
        <span
          class="sc-gZMcBi kBJdBl"
        >
          bravo slack bot
        </span>
        <span
          class="sc-gqjmRU fTEWBa"
        >
          install our slack integration in few simple clicks
        </span>
      </div>
      <div
        class="sc-gzVnrw gQetWj"
      >
        <img
          class="sc-iwsKbI hHToOU"
          src="gift-icon.png"
        />
        <span
          class="sc-gZMcBi kBJdBl"
        >
          give a bravo
        </span>
        <span
          class="sc-gqjmRU fTEWBa"
        >
          select receiver, write your text, send
        </span>
      </div>
      <div
        class="sc-gzVnrw gQetWj"
      >
        <img
          class="sc-iwsKbI hHToOU"
          src="message-icon.png"
        />
        <span
          class="sc-gZMcBi kBJdBl"
        >
          receive a bravo
        </span>
        <span
          class="sc-gqjmRU fTEWBa"
        >
          get notification when someone sends you a bravo
        </span>
      </div>
    </div>
  </div>
</div>
`;
